src/XMakeTasks/Copy.cs

```diff
@@ -408,6 +409,20 @@ CopyFileWithState copyFile
                 _copiedFiles = new TaskItem[0];
                 return true;
             }
+            
+            if (_destinationFiles != null)
```

Comment:
I think we really need to have both conflict-by-name-collision-in-destination-folder and conflict-by-explicit-destination-path errors for this to be fully realized.

Fortunately, it looks like that's pretty easy. Move this block to after the call to `InitializeDestinationFiles`, and you can get the full list, including paths generated by unpacking `DestinationFolder`.

Please also add a test for that situation (copy two files with the same name from different folders into a destination folder and validate that the error arises).